File: 73-set-matrix-zeroes/set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        ios_base::sync_with_stdio(0);
        unordered_map<int,int> row,col;
        for(int i=0;i<matrix.size();i++){
            for(int j=0;j<matrix[0].size();j++){
                if(matrix[i][j]==0){
                    row[i]=1;
                    col[j]=1;
                }    
            }
        }
        for(int i=0;i<matrix.size();i++){
            for(int j=0;j<matrix[0].size();j++){
                if(row.find(i)!=row.end() || col.find(j)!=col.end()) matrix[i][j]=0;
            }
        }
    }
};

File: 73-set-matrix-zeroes/set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        row = {} 
        col = {}
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if matrix[i][j] == 0:
                    row[i] = 1
                    col[j] =1
        for i in range(len(matrix)):
            if i in row:
                for j in range(len(matrix[0])):
                    matrix[i][j] = 0
        for j in range(len(matrix[0])):
            if j in col:
                for i in range(len(matrix)):
                    matrix[i][j] = 0


        